crates/bevy_animation/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+#[derive(Clone, TypeUuid, Debug, Default)]
+#[uuid = "d81b7179-0448-4eb0-89fe-c067222725bf"]
+pub struct AnimationClip {
+    curves: HashMap<EntityPath, Vec<VariableCurve>>,
+    duration: f32,
+}
+
+impl AnimationClip {
+    #[inline]
+    /// Hashmap of the [`VariableCurve`]s per [`EntityPath`].
+    pub fn curves(&self) -> &HashMap<EntityPath, Vec<VariableCurve>> {
+        &self.curves
+    }
+
+    /// Add a [`VariableCurve`] to an [`EntityPath`].
+    pub fn add_curve_to_path(&mut self, path: EntityPath, curve: VariableCurve) {
+        // Update the duration of the animation by this curve duration if it's longer
+        self.duration = self
+            .duration
+            .max(*curve.keyframe_timestamps.last().unwrap_or(&0.0));
+        self.curves.entry(path).or_default().push(curve);
+    }
+}
+
+/// Animation controls
+#[derive(Component, Reflect)]
+#[reflect(Component)]
+pub struct AnimationPlayer {
+    paused: bool,
+    repeat: bool,
+    speed: f32,
+    elapsed: f32,
+    animation_clip: Handle<AnimationClip>,
+}
+
+impl Default for AnimationPlayer {
+    fn default() -> Self {
+        Self {
+            paused: false,
+            repeat: false,
+            speed: 1.0,
+            elapsed: 0.0,
+            animation_clip: Default::default(),
+        }
+    }
+}
+
+impl AnimationPlayer {
+    /// Start playing an animation, resetting state of the player
+    pub fn play_animation(&mut self, handle: Handle<AnimationClip>) -> &mut Self {
+        *self = Self {
+            animation_clip: handle,
+            ..Default::default()
+        };
+        self
+    }
+
+    /// Set the animation to repeat
+    pub fn repeat(&mut self) -> &mut Self {
+        self.repeat = true;
+        self
+    }
+
+    /// Pause the animation
+    pub fn pause(&mut self) {
+        self.paused = true;
+    }
+
+    /// Unpause the animation
+    pub fn play(&mut self) {
+        self.paused = false;
+    }
+
+    /// Is the animation paused
+    pub fn is_paused(&self) -> bool {
+        self.paused
+    }
+
+    /// Speed of the animation playback
+    pub fn speed(&self) -> f32 {
+        self.speed
+    }
+
+    /// Set the speed of the animation playback
+    pub fn set_speed(&mut self, speed: f32) -> &mut Self {
+        self.speed = speed;
+        self
+    }
+
+    /// Time elapsed playing the animation
+    pub fn elapsed(&self) -> f32 {
+        self.elapsed
+    }
+
+    /// Seek to a specific time in the animation
+    pub fn set_elapsed(&mut self, elapsed: f32) -> &mut Self {
+        self.elapsed = elapsed;
+        self
+    }
+}
+
+/// System that will play all animations, using any entity with a [`AnimationPlayer`]
+/// and a [`Handle<AnimationClip>`] as an animation root
+pub fn animation_player(
+    time: Res<Time>,
+    animations: Res<Assets<AnimationClip>>,
+    mut animated: Query<(Entity, &mut AnimationPlayer)>,
+    named: Query<&Name>,
```

Comment:
Generally we use "plural" forms for queries. To follow that pattern, I think `transformed` should be `transforms` and `animated` should be `animation_players` (or just `players` if you prefer brevity)